### PR TITLE
Use FullName instead of individual ErrorableTextInputs

### PR DESCRIPTION
**Pull request by akainic**
One TODO down, many more to go....

### CODE REVIEW

File: _health-care/_js/utils/validations.js [hunk 1]
```diff
@@ -94,7 +92,10 @@ function isValidVeteranAddress(data) {
...
 
 function isValidSpouseInformation(data) {
-  return (isBlank(data.spouseSocialSecurityNumber) || isValidSSN(data.spouseSocialSecurityNumber)) &&
+  return (isBlank(data.spouseFullName.first) || isValidName(data.spouseFullName.first)) &&
+      (isBlank(data.spouseFullName.middle) || isValidName(data.spouseFullName.middle)) &&
+      (isBlank(data.spouseFullName.last) || isValidName(data.spouseFullName.last)) &&
+      (isBlank(data.spouseSocialSecurityNumber) || isValidSSN(data.spouseSocialSecurityNumber)) &&
       ((isBlank(data.spouseDateOfBirth.day) && isBlank(data.spouseDateOfBirth.month) && isBlank(data.spouseDateOfBirth.year)) ||
       isValidDate(data.spouseDateOfBirth.day, data.spouseDateOfBirth.month, data.spouseDateOfBirth.year)) &&
       ((isBlank(data.dateOfMarriage.day) && isBlank(data.dateOfMarriage.month) && isBlank(data.dateOfMarriage.year)) ||
```
Comment (joesoftware) [inline _health-care/_js/utils/validations.js]: These validations are getting hard to read. Maybe a validator of `isEnoughName()` could shrink the validator a bit? Good enough for now though.

Comment (akainic) [inline _health-care/_js/utils/validations.js]: I agree, they are hard to read (and hard for me to write too). Not sure what the right solution is off the top of my head, but definitely merits more thought.

**Comment by joesoftware** [comment]
LGTM with a comment for later discussion.

**Comment by akainic** [comment]
Thanks Joe! I'll create a task for us to revisit that issue later. Going to merge for now.

**Pull request merged**